src/com/esotericsoftware/kryonet/TcpConnection.java

```diff
@@ -46,10 +46,14 @@
 	private volatile long lastWriteTime, lastReadTime;
 	private int currentObjectLength;
 	private final Object writeLock = new Object();
+	private final int objectBufferSize;
+	private final int lengthLength;
 
 	public TcpConnection (Serialization serialization, int writeBufferSize, int objectBufferSize) {
 		this.serialization = serialization;
-		writeBuffer = ByteBuffer.allocate(writeBufferSize);
+		this.objectBufferSize = objectBufferSize;
+		lengthLength = serialization.getLengthLength();
+		writeBuffer = ByteBuffer.allocate(Math.max(writeBufferSize, lengthLength + objectBufferSize));
```

Comment:
Because bufferSize >= lengthLength + objectSize must be true in all cases.
If it is not so, user has entered incorrect values. We can react on that two ways:
first - throw an exception, second - use max(bufferSize, lengthLength + objectSize).
We have chosen second one, but we can change to exception.